app/views/Project/Tagging/Export/LeadsSelection/index.tsx

```diff
@@ -0,0 +1,354 @@
+                        mergedAs {
+                            id
+                            title
+                        }
+                    }
+                    assignee {
+                        id
+                        displayName
+                    }
+                    source {
+                        mergedAs {
+                            id
+                            title
+                        }
+                        id
+                        url
+                        title
+                    }
+                    entriesCounts {
+                        total
+                    }
+                    leadPreview {
+                        pageCount
+                    }
+                    isAssessmentLead
+                }
+            }
+        }
+    }
+`;
+
+interface Props {
+    className?: string;
+    projectId: string;
+    filterOnlyUnprotected: boolean;
+    hasAssessment?: boolean;
+    onSelectLeadChange: (values: string[]) => void;
+    selectedLeads: string[];
+    selectAll: boolean;
+    onSelectAllChange: (v: boolean) => void;
+    filterValues: Omit<SourceFilterOptionsQueryVariables, 'projectId'>;
+    onFilterApply: (value: Omit<SourceFilterOptionsQueryVariables, 'projectId'>) => void;
+}
+
+function LeadsSelection(props: Props) {
+    const {
+        projectId,
+        className,
+        selectedLeads,
+        onSelectLeadChange,
+        selectAll,
+        onSelectAllChange,
+        filterValues,
+        onFilterApply,
+        filterOnlyUnprotected,
+        hasAssessment,
+    } = props;
+
+    const sortState = useSortState();
+    const { sorting } = sortState;
+    const validSorting = sorting || defaultSorting;
+    const ordering = validSorting.direction === 'Ascending'
+        ? validSorting.name
+        : `-${validSorting.name}`;
+
+    const [activePage, setActivePage] = useState<number>(1);
+
+    const variables = useMemo(
+        (): ProjectSourceListQueryVariables | undefined => (
+            (projectId) ? {
+                ...filterValues,
+                projectId,
+                page: activePage,
+                pageSize: maxItemsPerPage,
+                ordering,
+            } : undefined
+        ),
+        [projectId, activePage, ordering, filterValues],
+    );
+
+    const {
+        data: projectSourcesResponse,
+        loading: projectSourcesPending,
+    } = useQuery<ProjectSourceListQuery, ProjectSourceListQueryVariables>(
+        PROJECT_LEADS,
+        {
+            skip: isNotDefined(variables),
+            variables,
+        },
+    );
+
+    const handleSelectAll = useCallback((value: boolean) => {
+        onSelectAllChange(value);
+        if (value) {
+            onSelectLeadChange([]);
+        } else {
+            // NOTE no leads selected in this case. If we set []
+            // all the leads will be selected which we don't want.
+            onSelectLeadChange(['-1']);
+        }
```

Comment:
This doesn't look correct.

Reply:
@tnagorra @thenav56 suggested we do this.